tests/tools/onert_train/src/nnfw_util.cc

```diff
@@ -46,4 +49,71 @@ uint64_t bufsize_for(const nnfw_tensorinfo *ti)
   return elmsize[ti->dtype] * num_elems(ti);
 }
 
+std::ostream &operator<<(std::ostream &os, const NNFW_TRAIN_OPTIMIZER &opt)
+{
+  switch (opt)
+  {
+    case NNFW_TRAIN_OPTIMIZER_ADAM:
+      os << "adam";
+      break;
+    case NNFW_TRAIN_OPTIMIZER_SGD:
+      os << "sgd";
+      break;
+    default:
+      os << "unsupported optimzier";
+  }
+  return os;
+}
+
+std::ostream &operator<<(std::ostream &os, const NNFW_TRAIN_LOSS &loss)
+{
+  switch (loss)
+  {
+    case NNFW_TRAIN_LOSS_MEAN_SQUARED_ERROR:
+      os << "mean squared error";
+      break;
+    case NNFW_TRAIN_LOSS_CATEGORICAL_CROSSENTROPY:
+      os << "categorical crossentropy";
+      break;
+    default:
+      os << "unsupported loss";
+  }
+  return os;
+}
+
+std::ostream &operator<<(std::ostream &os, const NNFW_TRAIN_LOSS_REDUCTION &loss_reduction)
+{
+  switch (loss_reduction)
+  {
+    case NNFW_TRAIN_LOSS_REDUCTION_INVALID:
+      os << "invalid (use default)";
```

Comment:
```suggestion
      os << "use default setting (model parameter or ONERT train setting)";
```

Reply:
I tried it, but it looks too long to display.

I append my opinion about how to print the `INVALID` option here - https://github.com/Samsung/ONE/pull/12296#discussion_r1429608402 

So, How about merging this PR as-is, and trying to reconfigure it later? 
Since I also have a plan to introduce a default(invalid) enum type for other configurations (`NNFW_TRAIN_OPTIMIZER`, `NNFW_TRIAN_LOSS`), I think i could fix it together.

Reply:
Well, however this `invalid` option means using model parameter or ONERT training setting. If we leave a `invalid` log about this option, the user may think that this option is invalid. How about using `use default setting` only for display?

Reply:
>  If we leave a invalid log about this option, the user may think that this option is invalid. 

I agree with this point.
updated!